source/common/upstream/conn_pool_map.h

```diff
@@ -84,9 +84,10 @@ template <typename KEY_TYPE, typename POOL_TYPE> class ConnPoolMap {
   absl::flat_hash_map<KEY_TYPE, std::unique_ptr<POOL_TYPE>> active_pools_;
   Event::Dispatcher& thread_local_dispatcher_;
   std::vector<IdleCb> cached_callbacks_;
-  Common::DebugRecursionChecker recursion_checker_;
   const HostConstSharedPtr host_;
+  // Keep smaller fields near the end to reduce padding
   const ResourcePriority priority_;
+  Common::DebugRecursionChecker recursion_checker_;
```

Comment:
Is this because `DebugRecursionChecker` has only a single bool field?

Reply:
Does moving `recursion_checker_` reduce `sizeof` for the class it's in?

Reply:
Pre change: Size of ConnPoolMap: 104
Post change: Size of ConnPoolMap: 96